Tailwind redesign

rlogank opened the pull request:
## To-dos to merge this PR

For @rlogank

- [x] Improve proportions between celeb gallery and celeb profile picture at all breakpoints
- [x] Discuss and resolve Hollowverse logo size
- [x] Add left border to blockquotes
- [x] Fix Facts blockquote after global styles have been removed
- [x] Blockquotes in editorial and Facts should have the same color
- [x] Make Celeb Gallery name part of the clickable area
- [x] Discuss and resolve app width size
- [x] Add open/close icons to Google Search

For @msafi

- [x] Restore Google Search

rlogank committed:
init

rlogank committed:
restyled homepage with tailwindcss, responsive

rlogank committed:
fixed minor issues

msafi committed:
Add react debounce input

msafi committed:
Add new search scaffolding behind feature flag

rlogank committed:
celeb page redesign part 1

rlogank committed:
added a lovely top border

msafi committed:
Create kgCall

rlogank committed:
added some fixes and tweaks

rlogank committed:
updated font + couple fixes

rlogank committed:
welcome

msafi committed:
Add prettier and undo unintended prettier formatting changes

msafi commented:
@rlogank This looks great dude!

I pushed a commit to undo some unintended formatting changes due to prettier and also added `.prettierrc.json` to prevent any more of these.

I'll ping you on Discord to discuss this further.

msafi committed:
before trying search route

msafi committed:
initial search 100% complete

msafi committed:
Knowledge Graph Celeb page 70% done

msafi committed:
Log missing celebs to Google Analytics

rlogank committed:
Removed SCSS and MUI, adapted all affected components, added accessibility features

msafi committed:
Merge branch 'search' into tw-redesign

# Conflicts:
#	lib/[celeb]/Facts/Fact/styles.module.scss
#	lib/[celeb]/index.tsx
#	lib/_app/AppBar/AppBar.tsx
#	lib/_app/index.tsx
#	lib/index/index.tsx
#	lib/~/contribute/index.tsx

msafi committed:
Fix some merge issues

rlogank committed:
added footer section, modified a blockquote

msafi committed:
Make Logan's signature higher contrast and reduce click surface

msafi committed:
use isEmpty to check arrays on celeb page

msafi committed:
Change `div` to `a` because that's more semantic. Using `div` was also breaking the CMD+click to open in new window feature.

Also, change the text from HOLLOWVERSE to Hollowverse then add `uppercase` class to achieve the visual effect because that's more semantically correct.

msafi committed:
Add recommended extensions so that VS Code users will get Extension recommendations when they open this project in VS Code.

msafi committed:
Trim bloat caused by empty `className`

msafi committed:
Create a reusable Card component to avoid repeating code

msafi committed:
The styles that are in globalStyles.css are mostly meant for the Editorial, but the way they were written they were affecting the entire app. I moved them to a CSS module and scoped them to only the Editorial section.

msafi committed:
Create a reusable Tag component

msafi committed:
Create a reusable lovely top border

msafi committed:
Clean up CelebGallery

msafi committed:
Use the Card component to display Facts

msafi committed:
Remove separator component

msafi committed:
Improve markup semantics on the contribute page

msafi committed:
Delete unused Icon component

msafi committed:
Miscellaneous cleanup

msafi committed:
Improve markup semantics in Fact

msafi committed:
Miscellaneous cleanup

msafi commented on lib/_app/AppBar/AppBar.tsx:
```diff
@@ -1,63 +1,32 @@
+        <div className="NAV mx-auto flex w-full max-w-5xl items-center justify-between">
+          <Link passHref href="/">
+            <a className="NAV-LOGO cursor-pointer select-none bg-gradient-to-r from-emerald-400 via-blue-400 to-purple-400 bg-clip-text font-extrabold uppercase tracking-tight text-transparent sm:text-xl lg:text-2xl">
+              Hollowverse
```
It seems that the nav bar height is consistent through out all breakpoints, but the Hollowverse text keeps shrinking as you go to lower breakpoints. Since the navbar height is static, I suggest we make the Hollowverse text static as well.

msafi commented on lib/[celeb]/Facts/Fact/index.tsx:
```diff
@@ -1,55 +1,61 @@
-            <blockquote className={s.content}>
-              <Typography>{value.quote}</Typography>
+            <blockquote className="mb-0 block border-l-4 border-blue-400 bg-blue-50">
+              {value.quote}
```
These blockquotes were being affected by the styles in `globalStyles.css`. I moved those styles and made them scoped to the Editorial only, but now these blockquotes don't look as good.

And could you add the left border back on both the Editorial and the Facts blockquotes?

msafi commented on lib/components/CelebGallery/index.tsx:
```diff
@@ -1,50 +1,49 @@
+              aria-label="Celebrity"
+              className="relative bottom-12 left-3 z-[9999] inline-flex bg-black bg-opacity-75 p-1.5 px-3 text-sm font-medium text-white lg:text-base"
             >
               {celebData.name}
```
Can you make this part of the clickable area?

msafi committed:
Reduce app width

msafi commented on lib/[celeb]/index.tsx:
```diff
@@ -7,23 +8,37 @@ import { Md } from '~/lib/[celeb]/Md/Md';
 
-      <TopSection />
+      <main className="flex flex-col bg-gray-100">
+        <div className="mx-auto max-w-4xl">
```
I think the app looks better at 4xl max than 5xl max. Let's discuss it if you disagree.

msafi committed:
Refactor directory structure in lib/

msafi committed:
Refactor api/ and ~contribute

msafi committed:
Override default `lg` breakpoint and...

Set it to `896px` to match `max-w-4xl`

msafi committed:
Add Google Search

msafi committed:
Fix style issue

msafi committed:
Fix AppBar

msafi committed:
Use next/link on Homepage

rlogank committed:
Completed pre-merge todo list

msafi committed:
Fix breakpoint setting

msafi committed:
Clean up card component

msafi committed:
Clean up classes in TopSection

msafi committed:
Optimize images

msafi committed:
Add accessibility linting

msafi committed:
Clean up unnecessary classes

msafi committed:
Reverse some removed classes
